### ISSUE TITLE
[MD007] ignores indent config

### ISSUE THREAD

**Issue by FBruzzesi** [rvben/rumdl#210]
As of the latest version `rumdl 0.0.197`, the `indent` argument of MD007 seems to be ignored. Is this a regression?

**.rumdl.toml**
```toml
[global]
line-length = 120

[MD007]
indent = 4
start-indented = false
```

**repro.md**

```md
# Title

* some
    * list
    * items

```

```terminal
$ uv tool run rumdl version && uv tool run rumdl check .
rumdl 0.0.197
repro.md:6:1: [MD007] Expected 2 spaces for indent depth 1, found 4 [*]
repro.md:7:1: [MD007] Expected 2 spaces for indent depth 1, found 4 [*]

Issues: Found 2 issues in 1 file (21ms)
Run `rumdl fmt` to automatically fix 2 of the 2 issues
```

In previous version that's not ignored:

```terminal
$ uv tool install rumdl==0.0.196
Resolved 1 package in 2ms
Installed 1 package in 3ms
 + rumdl==0.0.196
Installed 1 executable: rumdl

$ uv tool run rumdl version && uv tool run rumdl check .
rumdl 0.0.196

Success: No issues found in 1 file (23ms)
```

**Comment by chrisgrieser** [comment]
I also noticed this

**Comment by chrisgrieser** [comment]
Seems to be fixed in 0.0.198.

**Comment by FBruzzesi** [comment]
**Disclaimer**: I might be missing something to add in the config

I am still getting the issue in CI where I run rumdl https://github.com/FBruzzesi/anyschema/actions/runs/20411415944/job/58648795082?pr=85

**Comment by chrisgrieser** [comment]
Hmm, that's weird. I have the same config.

```toml
[MD007]  # ul-indent
indent = 4
```

I don't run it via `uv` though, I use the default GitHub action (` - uses: rvben/rumdl@v0`), and I get no errors in my CI: https://github.com/chrisgrieser/shimmering-obsidian/actions/runs/20411517268/job/58649030969

**Comment by martimlobao** [comment]
Still seeing this in v0.0.199

**Comment by martimlobao** [comment]
Though it might be some cache issue? My setting is also set to 4 spaces, but the errors I'm seeing are for 3 expected spaces, which used to be what I had. 🤔 

```
test.md:48:1: [MD007] Expected 3 spaces for indent depth 1, found 4 [*]
```

FWIW, i did try deleting the local `.rumdl_cache/` directory but still got the same error.

**Comment by chrisgrieser** [comment]
@martimlobao You could try running `rumdl config` to check if the config is loaded correctly

**Comment by FBruzzesi** [comment]
@chrisgrieser thanks for the suggestion. For me the output seems to be as expected:

```terminal
$ uv tool run rumdl config | grep pyproject
disable = ["MD031", "MD046"]                              [from pyproject.toml]
include = ["docs/**", "README.md", "AGENTS.md"]           [from pyproject.toml]
flavor = MkDocs                                           [from pyproject.toml]
indent = 4                                                [from pyproject.toml]
start-indented = false                                    [from pyproject.toml]
line-length = 120                                         [from pyproject.toml]
```

---

> I don't run it via uv though, I use the default GitHub action ( - uses: rvben/rumdl@v0), and I get no errors in my CI: https://github.com/chrisgrieser/shimmering-obsidian/actions/runs/20411517268/job/58649030969

This is also a good suggestion - for me personally using uv makes it easy to run it locally before committing (it's super fast anyway :))

**Comment by martimlobao** [comment]
@chrisgrieser I did, the indentation setting is set to 4 spaces.

**Comment by martimlobao** [comment]
My suspicion that this is related to some cache just increased, now I don't get any errors anymore.

**Comment by chrisgrieser** [comment]
Well now I do get those errors 🙈 

https://github.com/chrisgrieser/obsidian-proofreader/actions/runs/20444982360/job/58746397333#step:3:1

**Comment by rvben** [comment]
Fixed. The `indent` configuration is now respected correctly. Thanks @FBruzzesi for reporting!

**Comment by FBruzzesi** [comment]
Hey @rvben thanks for the quick support. I can still see the issue in CI with rumdl 0.0.204

- Failing [Github action](https://github.com/FBruzzesi/anyschema/actions/runs/20539962808/job/59002941542?pr=97)
- Rumdl [configuration](https://github.com/FBruzzesi/anyschema/blob/83d472539ee9e7532e4e5e2249d392c59d55b853/pyproject.toml#L190-L206)

As you can see from the logs, the correct config is picked up:

```terminal
Downloading cpython-3.14.2-linux-x86_64-gnu (download) (33.7MiB)
 Downloaded cpython-3.14.2-linux-x86_64-gnu (download)
Downloading rumdl (3.2MiB)
 Downloaded rumdl
Installed 1 package in 2ms
rumdl 0.0.204
[global]                                        
disable = ["MD031", "MD046"]                    [from pyproject.toml]
include = ["docs/**", "README.md", "AGENTS.md"] [from pyproject.toml]
flavor = MkDocs                                 [from pyproject.toml]
line_length = 120                               [from pyproject.toml]
[MD007]                                         
indent = 4                                      [from pyproject.toml]
start-indented = false                          [from pyproject.toml]
[MD013]                                         
line-length = 120                               [from pyproject.toml]
```

**Comment by redactedscribe** [comment]
Hi @rvben. Can you check whether this is still working?

I have:

```toml
[ul-indent]
indent = 4
style = "fixed"
```

And Zed is complaining "Expected 4 spaces for indent depth 1, found 2". Format on save is working, but it keeps on forcing 2 spaces instead of the configured 4.

Test data:

```text
- Bullet item
  - Nested bullet (4-space fixed indent - uses your config)
  1. Ordered child
     - Bullet under ordered (text-aligned at column 11, not fixed)
```

On a side note, I don't see [the warning](https://rumdl.dev/md007/#do-what-i-mean-behavior) that should display when defining an indent + using text-aligned. Maybe the above bug masks it.

rumdl 0.1.26

**Comment by rvben** [comment]
Hi @redactedscribe, I investigated this and can't reproduce the issue you describe. Are there more formatters chained in the Zed config that might conflict?

**Comment by redactedscribe** [comment]
> Hi [@redactedscribe](https://github.com/redactedscribe), I investigated this and can't reproduce the issue you describe. Are there more formatters chained in the Zed config that might conflict?

There is only 1 other formatter and it's for GDScript language, and so I don't think that's chained in any way. Also, I'm using Zed on Windows which can be a bit more buggy than on other OSes.

Zed config:

```json
	"languages": {
		"Markdown": {
			"code_actions_on_format": {
				"source.fixAll.rumdl": true,
			},
			"format_on_save": "on",
			"language_servers": ["rumdl"],
		},
	},
```

I isolated the rumdl config to just the `[ul-indent]` rules and the markdown file to just the test data and it made no difference. If I notice something else I'll let you know.

**Comment by rvben** [comment]
Thanks @redactedscribe for the additional info. I'll try to replicate and see what might be going wrong.

**Comment by rvben** [comment]
@redactedscribe I found and fixed several LSP-side bugs that could contribute (config cache not invalidating properly, `source.fixAll` not appearing with narrow cursor ranges). These will be in the next release.

However, the most likely root cause is Zed's formatter pipeline. Your config doesn't set `formatter`, so it defaults to `"auto"` which can run Prettier after rumdl's fix, normalizing indentation back to 2 spaces.

Could you try adding this to your Markdown config?

```json
"formatter": { "language_server": { "name": "rumdl" } }
```

**Comment by redactedscribe** [comment]
> Could you try adding this to your Markdown config?
> 
> "formatter": { "language_server": { "name": "rumdl" } }

I updated my rumdl to the latest available 0.1.28 via Scoop and added this to my Zed config. Without it, it won't respect `indent = 4`, but with it it does.

I was questioning if I had originally mistakenly not opened the working folder in Zed (so presumably it would only be formatting with default settings and ignoring my rumdl config), but now that I know I have and since the added Zed configuration is toggling the desired behaviour, I guess this really was the fix.

Thanks!
